dashboard/src/components/users/EditUserForm/EditUserForm.tsx

```diff
@@ -0,0 +1,472 @@
+    .required('This field is required.'),
+  emailVerified: Yup.boolean().optional(),
+  phoneNumber: Yup.string().nullable(),
+  phoneNumberVerified: Yup.boolean().optional(),
+  locale: Yup.string(),
+  defaultRole: Yup.string(),
+  roles: Yup.array().of(
+    Yup.object({
+      role: Yup.string(),
+      value: Yup.bool(),
+    }),
+  ),
+});
+
+export type EditUserFormValues = Yup.InferType<
+  typeof EditUserFormValidationSchema
+>;
+
+export default function EditUserForm({
+  user,
+  onEditUser,
+  onCancel,
+  onBanUser,
+  onDeleteUser,
+}: EditUserFormProps) {
+  const { onDirtyStateChange, openDialog } = useDialog();
+  const { currentApplication } = useCurrentWorkspaceAndApplication();
+
+  const { data } = useGetRolesQuery({
+    variables: { id: currentApplication.id },
+  });
+
+  const allAvailableProjectRoles = useMemo(
+    () => getUserRoles(data?.app?.authUserDefaultAllowedRoles),
+    [data],
+  );
+
+  const isAnonymous = user.roles.some((role) => role.role === 'anonymous');
+
+  const form = useForm<EditUserFormValues>({
+    reValidateMode: 'onSubmit',
+    resolver: yupResolver(EditUserFormValidationSchema),
+    defaultValues: {
+      avatarURL: user.avatarUrl,
+      displayName: user.displayName,
+      email: user.email,
+      emailVerified: user.emailVerified,
+      phoneNumber: user.phoneNumber,
+      phoneNumberVerified: user.phoneNumberVerified,
+      locale: user.locale,
+      defaultRole: user.defaultRole,
+      roles: allAvailableProjectRoles?.map((role) => {
+        const userRole = user.roles.find((sr) => sr.role === role.name);
+        return { role: role.name, value: !!userRole };
+      }),
+    },
+  });
+
+  const {
+    register,
+    formState: { errors, dirtyFields },
+  } = form;
+
+  const isDirty = Object.keys(dirtyFields).length > 0;
+
+  useEffect(() => {
+    onDirtyStateChange(isDirty, 'dialog');
+  }, [isDirty, onDirtyStateChange]);
+
+  useEffect(() => {
+    form.reset(() => ({
+      avatarURL: user.avatarUrl,
+      displayName: user.displayName,
+      email: user.email,
+      emailVerified: user.emailVerified,
+      phoneNumber: user.phoneNumber,
+      phoneNumberVerified: user.phoneNumberVerified,
+      locale: user.locale,
+      defaultRole: user.defaultRole,
+      roles: allAvailableProjectRoles?.map((role) => {
+        const userRole = user.roles.find((sr) => sr.role === role.name);
+        return { role: role.name, value: !!userRole };
+      }),
+    }));
+  }, [user, form, allAvailableProjectRoles]);
+
+  function handleChangeUserPassword() {
+    openDialog('EDIT_USER_PASSWORD', {
+      title: 'Change Password',
+      payload: { user },
+      props: {
+        titleProps: { className: 'mx-auto' },
+        PaperProps: { className: 'max-w-md' },
+      },
+    });
+  }
+
+  const signInMethodsAvailable = user.email || user.userProviders.length !== 0;
+
+  return (
+    <FormProvider {...form}>
+      <Form
+        className="divide-y border-y"
```

Comment:
On larger screens, the form doesn't expand to full height. You have plenty of examples to take into consideration.

<img width="1000" alt="Screenshot 2022-12-19 at 9 59 51" src="https://user-images.githubusercontent.com/310881/208388004-df28d85e-119f-416a-a16d-b8486b9482d5.png">